### ISSUE TITLE
(custom-resources): use lambda.Runtime.NODEJS_18_X by default in Provider

### ISSUE THREAD

**Issue by twulf** [aws/aws-cdk#23682]
### Describe the feature

Similar to https://github.com/aws/aws-cdk/issues/19117 we should update to latest supported Node.js runtime for the custom-resources lambda prior to Node 14 going out of service.

https://github.com/aws/aws-cdk/blob/2d40d7713ddfedbcba41edfd67afab3035c5b1f2/packages/%40aws-cdk/custom-resources/lib/provider-framework/provider.ts#L210

### Use Case

All lambdas created by aws-cdk internally should be updated prior to EOL.

### Proposed Solution

Update to NODEJS_18_X when creating lambda in custom-resources.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.60.0

### Environment details (OS name and version, etc.)

MacOS 12.6.2

**Comment by wilhen01** [comment]
+1 for this. Node 14 EOL is now rapidly approaching!

**Comment by pahud** [comment]
Node18 has been [supported](https://aws.amazon.com/about-aws/whats-new/2023/04/aws-lambda-node-js-18-govcloud-regions/) in GovCloud and AWS China regions as well. I believe we can create a PR for this now.

**Comment by cornelcroi** [comment]
Hi,

Any updates on this?
I have an application impossible to deploy now.
Any workaround possible?
Thanks.

**Comment by kaizencc** [comment]
@cornelcroi we are working to move to Node 18 but your application should not be impossible to deploy. If you are seeing errors related to Node 12, then you need to update your aws-cdk-lib version. aws-cdk-lib exclusively uses Node 14+ currently.

**Comment by anoppe** [comment]
It might not be the most elegant solution, but If you don't want to be blocked by CDK lib releases, you could also use an aspect to change the runtime of the lambda created by, for example, the provider framework.
A naive example, written in Java: 
```java
// Change the provider framework lambda runtime to node18.
Aspects.of(this).add(node -> {
    if (node instanceof CfnFunction) {
        if (((CfnFunction) node).getRuntime().startsWith("node")
                && !((CfnFunction) node).getRuntime().equals("nodejs18.x")) {
            ((CfnFunction) node).setRuntime("nodejs18.x");
        }
    }
});
```

**Comment by peterwoodworth** [comment]
We did this, but we have missed some. Closing this in favor of the bug report that's more up to date https://github.com/aws/aws-cdk/issues/26461
